File: nim/helloworld/helloworld.nim
import gintro/[gtk, gobject, gio]


proc buttonClicked (button: Button) =
  quit(QuitSuccess)


proc appActivate (app: Application) =
  let window = newApplicationWindow(app)
  window.title = "Hello World"
  window.defaultSize = (250, 50)
  let button = newButton("Hello World")
  window.add(button)
  button.connect("clicked",  buttonClicked)
  window.showAll


proc main =
  let app = newApplication("org.gtk.HelloWorld")
  connect(app, "activate", appActivate)
  discard app.run


main()

File: lua/helloworld/helloworld.lua
#!/usr/bin/env lua


local lgi = require "lgi"
local Gtk = lgi.require("Gtk", "3.0")
local Gio = lgi.Gio


local app = Gtk.Application.new(
   "org.gtk.HelloWorld",
   Gio.ApplicationFlags.HANDLES_OPEN
)


local function show_app()
   local window = Gtk.Window {
      type = Gtk.WindowType.TOPLEVEL,
      default_width = 300,
      default_height = 300,
      application = app,
      title = "Hello World",

      child = Gtk.ToolButton {
         id = "hello_world",
         label = "Hello World!",
         on_clicked = function() app:quit() end,
      }
   }
   window:show_all()
   return window
end


function app:on_activate()
   show_app()
end


return app:run {}
